src/strategy/exponential_backoff.rs

```diff
@@ -1,14 +1,16 @@
 use std::time::Duration;
 use std::iter::Iterator;
-use std::u64::{MAX as U64_MAX};
+use std::u64::MAX as U64_MAX;
 
 /// A retry strategy driven by exponential back-off.
 ///
 /// The power corresponds to the number of past attempts.
-#[derive(Clone)]
+#[derive(Debug, Clone)]
 pub struct ExponentialBackoff {
     current: u64,
-    base: u64
+    base: u64,
+    factor: Option<u64>,
```

Comment:
Without benchmarks showing an improvement, and expecting only a minimal difference (if any at all -- an additional branch could also make performance worse), I think erring on the side of the simpler code here is the right call to make.